subsys/bluetooth/mesh/light_xyl_srv.c

```diff
@@ -0,0 +1,552 @@
+	struct bt_mesh_light_xy_status status = { 0 };
+	struct bt_mesh_lightness_status light;
+
+	srv->lightness_srv.handlers->light_get(&srv->lightness_srv, ctx,
+					       &light);
+	srv->handlers->xy_get(srv, NULL, &status);
+
+	struct bt_mesh_light_xyl_status xyl_status = {
+		.params = { .xy = status.current, .lightness = light.current },
+		.remaining_time = status.remaining_time,
+	};
+
+	xyl_rsp(model, ctx, &xyl_status, BT_MESH_LIGHT_XYL_STATUS);
+}
+
+static void xyl_set_handle(struct bt_mesh_model *model,
+			   struct bt_mesh_msg_ctx *ctx,
+			   struct net_buf_simple *buf)
+{
+	xyl_set(model, ctx, buf, true);
+}
+
+static void xyl_set_unack_handle(struct bt_mesh_model *model,
+				 struct bt_mesh_msg_ctx *ctx,
+				 struct net_buf_simple *buf)
+{
+	xyl_set(model, ctx, buf, false);
+}
+
+static void target_get_handle(struct bt_mesh_model *model,
+			      struct bt_mesh_msg_ctx *ctx,
+			      struct net_buf_simple *buf)
+{
+	if (buf->len != BT_MESH_LIGHT_XYL_MSG_LEN_GET) {
+		return;
+	}
+
+	struct bt_mesh_light_xyl_srv *srv = model->user_data;
+	struct bt_mesh_light_xy_status status = { 0 };
+	struct bt_mesh_lightness_status light;
+
+	srv->lightness_srv.handlers->light_get(&srv->lightness_srv, ctx,
+					       &light);
+	srv->handlers->xy_get(srv, NULL, &status);
+
+	struct bt_mesh_light_xyl_status xyl_status = {
+		.params.xy = status.target,
+		.params.lightness = light.target,
+		.remaining_time = status.remaining_time,
+	};
+
+	xyl_rsp(model, ctx, &xyl_status, BT_MESH_LIGHT_XYL_TARGET_STATUS);
+}
+
+static void default_encode_status(struct bt_mesh_light_xyl_srv *srv,
+				  struct net_buf_simple *buf)
+{
+	bt_mesh_model_msg_init(buf, BT_MESH_LIGHT_XYL_DEFAULT_STATUS);
+	net_buf_simple_add_le16(buf, srv->lightness_srv.default_light);
+	net_buf_simple_add_le16(buf, srv->xy_default.x);
+	net_buf_simple_add_le16(buf, srv->xy_default.y);
+}
+
+static void default_rsp(struct bt_mesh_model *model,
+			struct bt_mesh_msg_ctx *rx_ctx)
+{
+	struct bt_mesh_light_xyl_srv *srv = model->user_data;
+
+	BT_MESH_MODEL_BUF_DEFINE(msg, BT_MESH_LIGHT_XYL_DEFAULT_STATUS,
+				 BT_MESH_LIGHT_XYL_MSG_LEN_DEFAULT);
+	default_encode_status(srv, &msg);
+	(void)bt_mesh_model_send(model, rx_ctx, &msg, NULL, NULL);
+}
+
+static void default_set(struct bt_mesh_model *model,
+			struct bt_mesh_msg_ctx *ctx, struct net_buf_simple *buf,
+			bool ack)
+{
+	if (buf->len != BT_MESH_LIGHT_XYL_MSG_LEN_DEFAULT) {
+		return;
+	}
+
+	struct bt_mesh_light_xyl_srv *srv = model->user_data;
+	uint16_t light = repr_to_light(net_buf_simple_pull_le16(buf), ACTUAL);
+
+	srv->xy_default.x = net_buf_simple_pull_le16(buf);
+	srv->xy_default.y = net_buf_simple_pull_le16(buf);
+	srv->lightness_srv.default_light = light;
+	store_state(srv);
+
+	if (srv->handlers->default_update) {
+		struct bt_mesh_light_xy xy_set = {
+			.x = srv->xy_default.x,
+			.y = srv->xy_default.y,
+		};
+
+		srv->handlers->default_update(srv, ctx, &xy_set);
```

Comment:
The `default_update` and `range_update` functions have `old` and `new` parameters. Can also just pass `&srv->xy_default` directly